### PR TITLE
useOnScrollBottom Hooks for fetch on scroll event

### PR DESCRIPTION
**Pull request by pineshmenat**

### CODE REVIEW

**Comment by pineshmenat** [comment]
> can you please fix the conflicts.

done

File: app/util/api.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ export function fetchNewsIDsByType(type = 'top') {
     return fetch(`https://hacker-news.firebaseio.com/v0/${type}stories.json?print=pretty`)
         .then((res) => res.json())
         .then((ids) => {
-            return ids.slice(0, 50);
+            return ids;
```
Comment (AmitMirgal) [inline app/util/api.js:5]: same as above,,, no need of return keyword

File: app/util/helpers.js [hunk 2]
```diff
@@ -1,6 +1,40 @@
...
 export function formatDate(time) {
     return new Date(time * 1000).toLocaleDateString("en-US", {
         hour: "numeric",
         minute: "numeric"
     })
-}
\ No newline at end of file
+}
+
+export function getDocHeight() {;
```
Comment (AmitMirgal) [inline app/util/helpers.js:34]: accidentally **semicolon** got added...

File: app/util/helpers.js [hunk 3]
```diff
@@ -1,6 +1,40 @@
+import React, {useState, useEffect} from 'react';
+import debounce from "lodash.debounce";
+
+export function useOnScrollBottom(error, isLoading, allIDs) {
```
Comment (AmitMirgal) [inline app/util/helpers.js:4]: i think helper is more tends to manipulation/operations (it holds utility functionality), here i can see the react code as well so can we extract the react code and put it somewhere....

File: app/components/NewsList.js [hunk 4]
```diff
@@ -1,14 +1,15 @@
 import React, {useState, useEffect} from 'react'
 import News from './News'
 import {fetchNewsIDsByType} from '../util/api.js'
+import {useOnScrollBottom} from '../util/helpers';
 
-export default function NewsList(props){
+function useFetchNewsIDs(type) {
```
[the remark this answers is not in the document]
Comment (pineshmenat) [reply app/components/NewsList.js]: done, thanks for suggestion

File: app/components/NewsList.js [hunk 5]
```diff
@@ -17,15 +18,21 @@ export default function NewsList(props){
...
 
+export default function NewsList(props){
+    const {newsIDs, isLoading, error} = useFetchNewsIDs(props.type);
+    const {limitedIDs, hasMore} = useOnScrollBottom(error, isLoading, newsIDs);
     return (
         <ul>
            {isLoading === true ? <p>Loading...</p> : (error !== null ? <p>Error!</p> :
-               newsIDs.map((id) => {
+               limitedIDs.map((id) => {
                    return <News newsID={id} key={id}/>
```
[the remark this answers is not in the document]
Comment (pineshmenat) [reply app/components/NewsList.js]: done, thanks

File: app/util/api.js (hunk 1 above)
Comment (pineshmenat) [reply app/util/api.js]: done

File: app/util/helpers.js (hunk 2 above)
Comment (pineshmenat) [reply app/util/helpers.js]: 😲 Thanks for catching it, don't know how even with this ; code is working fine

File: app/util/helpers.js (hunk 3 above)
Comment (pineshmenat) [reply app/util/helpers.js]: yeah, moved to custom-hooks

**Comment by pineshmenat** [comment]
all good now?

File: app/custom-hooks/custom-hooks.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
```
Comment (AmitMirgal) [inline app/custom-hooks/custom-hooks.js:1]: what i mean was 
cutom-hooks (folder)
inside the folder
1. useFetchNewsIDs.js (1st file)
2. useOnScrollBottom.js (2nd file)
so in total custom-hooks has 2 different files

File: app/util/api.js [hunk 7]
```diff
@@ -1,33 +1,25 @@
 export function fetchNewsIDsByType(type = 'top') {
     return fetch(`https://hacker-news.firebaseio.com/v0/${type}stories.json?print=pretty`)
         .then((res) => res.json())
-        .then((ids) => {
-            return ids;
-        })
+        .then((ids) => ids);
```
Comment (AmitMirgal) [inline app/util/api.js:4]: no need of round brackets for parameter, but i am ok to keep as it is... upto you..

File: app/util/api.js [hunk 8]
```diff
@@ -1,33 +1,25 @@
...
+        .then((comment) => comment);
 }
 
 export function fetchPostsByUser(userId) {
     return fetch(`https://hacker-news.firebaseio.com/v0/user/${userId}.json?print=pretty`)
     .then((res) => res.json())
-    .then((userData) => {
-        return userData;
-    });
+    .then((userData) => userData);
```
Comment (AmitMirgal) [inline app/util/api.js:24]: same as above for all parameter

Comment (pineshmenat) [reply app/util/api.js]: yeah it looks more readable, thanks

File: app/util/api.js (hunk 7 above)
Comment (pineshmenat) [reply app/util/api.js]: yeah it looks more readable, thanks

File: app/custom-hooks/custom-hooks.js (hunk 6 above)
Comment (pineshmenat) [reply app/custom-hooks/custom-hooks.js]: yeah it's a good idea, thanks

**Comment by pineshmenat** [comment]
I am merging now,  thanks for your time Amit

**Pull request merged**